shared/src/state-migrations/43-generalize-materials-personnel.ts

```diff
@@ -0,0 +1,206 @@
+    [Key in string]: string;
+} = {
+    gf: 'GF',
+    notarzt: 'NA',
+    notSan: 'NFS',
+    rettSan: 'RS',
+    san: 'San',
+};
+
+export const personnelTypeNames: {
+    [key in string]: string;
+} = {
+    gf: 'Gruppenführer',
+    notarzt: 'Notarzt',
+    notSan: 'Notfallsanitäter',
+    rettSan: 'Rettungssanitäter',
+    san: 'Sanitäter',
+};
+export const generalizeMaterialsPersonnel43: Migration = {
+    action: (_, action) => {
+        const actionType = (action as { type: string }).type;
+
+        if (actionType === '[Vehicle] Add vehicle') {
+            const typedAction = action as {
+                vehicleParameters: VehicleParameters;
+            };
+            typedAction.vehicleParameters.materials.forEach((material) => {
+                const materialType: MaterialType =
+                    material.image.url === '/assets/material.svg'
+                        ? 'standard'
+                        : 'big';
+                material.baseTemplateId = materialTemplateIds[materialType];
+                material.name = materialTypeNames[materialType];
+            });
+            typedAction.vehicleParameters.personnel.forEach((personnel) => {
+                personnel.name = personnelTypeNames[personnel.personnelType];
+                personnel.baseTemplateId =
+                    personnelTemplateIds[personnel.personnelType];
+                personnel.abbreviation =
+                    personnelTypeAbbreviations[personnel.personnelType];
+            });
+        } else if (
+            actionType === '[Emergency Operation Center] Send Alarm Group'
+        ) {
+            const typedAction = action as {
+                sortedVehicleParameters: VehicleParameters[];
+            };
+            Object.values(typedAction.sortedVehicleParameters).forEach(
+                (vehicleParameters) =>
+                    vehicleParameters.materials.forEach((material) => {
+                        const materialType: MaterialType =
+                            material.image.url === '/assets/material.svg'
+                                ? 'standard'
+                                : 'big';
+                        material.baseTemplateId =
+                            materialTemplateIds[materialType];
+                        material.name = materialTypeNames[materialType];
+                    })
+            );
+            Object.values(typedAction.sortedVehicleParameters).forEach(
+                (vehicleParameters) =>
+                    vehicleParameters.personnel.forEach((personnel) => {
+                        personnel.name =
+                            personnelTypeNames[personnel.personnelType];
+                        personnel.baseTemplateId =
+                            personnelTemplateIds[personnel.personnelType];
+                        personnel.abbreviation =
+                            personnelTypeAbbreviations[personnel.personnelType];
+                    })
+            );
```

Comment:
This code is largely identical to the code for the add vehicle action. Could you extract common logic into a function? (For example, extracting the bodies of the inner `.forEach()` loops into a `migrateMaterial` (or `migratePersonnel`) function)